contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "../interfaces/OCR2DRClientInterface.sol";
+import "../interfaces/OCR2DROracleInterface.sol";
+
+/**
+ * @title OCR2DR oracle contract (stub for now)
+ */
+contract OCR2DROracle is OCR2DROracleInterface {
+  event OracleRequest(address sender, uint256 nonce, bytes data);
+  event CancelOracleRequest(bytes32 indexed requestId);
+  event OracleResponse(bytes32 indexed requestId);
+
+  function sendRequest(
+    address sender,
+    uint256 nonce,
+    uint256, /* subscriptionId */
+    bytes calldata data
+  ) external override {
+    emit OracleRequest(sender, nonce, data);
+  }
+
+  function cancelRequest(bytes32 requestId) external override {
+    emit CancelOracleRequest(requestId);
+  }
+
+  function fulfillRequest(
+    bytes32 requestId,
+    address callbackAddress,
+    bytes calldata response,
+    bytes calldata err
+  ) external override {
+    OCR2DRClientInterface client = OCR2DRClientInterface(callbackAddress);
```

Comment:
Again, verify the caller?

Reply:
Correct. The next piece of work is the Oracle and we should address all you mentioned in Oracle implementation.